gw_eccentricity/load_data.py

```diff
@@ -827,22 +834,24 @@ def load_sxs_catalogformat(**kwargs):
                 "This is required to get the binary parameters which are "
                 "used to eavaluate the zero ecc waveform.")
 
-    # load modes
-    modes_dict = {}
-    data = h5py.File(filepath, "r")
-    waveform_data = data[f"Extrapolated_N{extrap_order}.dir"]
-    for idx, mode in enumerate(mode_array):
-        ell, m = mode
-        mode_data = waveform_data[f"Y_l{ell}_m{m}.dat"]
-        # create the time array only once
-        if idx == 0:
-            time = mode_data[:, 0]
-            t = np.arange(time[0], time[-1], dt)
-        hlm = mode_data[:, 1] + 1j * mode_data[:, 2]
-        amp_interp = interpolate(t, time, np.abs(hlm))
-        phase_interp = interpolate(t, time, -np.unwrap(np.angle(hlm)))
-        hlm_interp = amp_interp * np.exp(-1j * phase_interp)
-        modes_dict.update({(ell, m): hlm_interp})
+    # Check file format
+    if "Strain_N" in kwargs["filepath"]:
+        # find the extrapolation order from file name and check if
+        # it is consistent with the value provided via kwargs.
+        # If they are not consistent, raise warnings and ignore
+        # the value provided in the kwargs.
+        base_name = os.path.basename(kwargs["filepath"])
+        extrap_order_from_filename = int(
+            re.findall(r"\d", base_name)[0])
```

Comment:
Just to be safer, search for the exact format. Something like
```
matches = re.findall("StrainN(\d).h5")
if len(matches) != 1:
    raise Exception...
extrap_order_from_filename = int(matches[0])
```

Reply:
Done